tests/e2e/models/CMakeLists.txt

```diff
@@ -146,3 +146,103 @@ iree_static_linker_test(
     "--iree-vm-target-index-bits=32"
   EMITC
 )
+
+if((IREE_TARGET_BACKEND_LLVM_CPU OR DEFINED IREE_HOST_BINARY_ROOT) AND
+   IREE_HAL_DRIVER_LOCAL_SYNC)
+  iree_bytecode_module(
+    NAME
+      iree_run_module_bytecode_module_llvm_cpu
+    SRC
+      "${IREE_ROOT_DIR}/tools/test/iree-run-module.mlir"
+    MODULE_FILE_NAME
+      "iree_run_module_bytecode_module_llvm_cpu.vmfb"
+    C_IDENTIFIER
+      "iree_tests_e2e_models_iree_run_module_bytecode_module_llvm_cpu"
+    FLAGS
+      "--iree-hal-target-backends=llvm-cpu"
+  )
+
+  iree_run_module_test(
+    NAME
+      iree_run_module_correctness_test
```

Comment:
I'm not sure this is the right directory for this sort of test of the tooling itself. If you put it under tools/test, would we still be able to run it on cross-compile?

Reply:
Yes we can. Previously I wasn't able to put it under `iree/tools`, but `iree/tools/test` works fine. Move the test there.